air/src/execution_step/boxed_value/jvaluable/cell_vec_resolved_call_result.rs

```diff
@@ -24,29 +24,37 @@ use crate::JValue;
 use crate::LambdaAST;
 use crate::SecurityTetraplet;
 
-use air_lambda_ast::format_ast;
-
 use std::borrow::Cow;
 use std::ops::Deref;
 
 impl JValuable for std::cell::Ref<'_, Vec<ValueAggregate>> {
-    fn apply_lambda<'i>(&self, lambda: &LambdaAST<'_>, exec_ctx: &ExecutionCtx<'i>) -> ExecutionResult<&JValue> {
+    fn apply_lambda<'i>(
+        &self,
+        lambda: &LambdaAST<'_>,
+        exec_ctx: &ExecutionCtx<'i>,
+    ) -> ExecutionResult<Cow<'_, JValue>> {
         let stream_iter = self.iter().map(|r| r.result.deref());
-        let select_result = select_from_stream(stream_iter, lambda, exec_ctx)?;
+        let select_result = select_by_lambda_from_stream(stream_iter, lambda, exec_ctx)?;
         Ok(select_result.result)
     }
 
     fn apply_lambda_with_tetraplets<'i>(
         &self,
         lambda: &LambdaAST<'_>,
         exec_ctx: &ExecutionCtx<'i>,
-    ) -> ExecutionResult<(&JValue, SecurityTetraplet)> {
+    ) -> ExecutionResult<(Cow<'_, JValue>, SecurityTetraplet)> {
         let stream_iter = self.iter().map(|r| r.result.deref());
-        let select_result = select_from_stream(stream_iter, lambda, exec_ctx)?;
+        let select_result = select_by_lambda_from_stream(stream_iter, lambda, exec_ctx)?;
 
-        let tetraplet = &self[select_result.tetraplet_idx].tetraplet;
-        let mut tetraplet = tetraplet.as_ref().clone();
-        tetraplet.add_lambda(&format_ast(lambda));
+        let tetraplet = match select_result.tetraplet_idx {
+            Some(idx) => {
+                let tetraplet = &self[idx].tetraplet;
+                let mut tetraplet = tetraplet.as_ref().clone();
+                tetraplet.add_lambda(&lambda.to_string());
```

Comment:
Can it happen that older tetraplet had `.$.field`, and then we add `.lambda` (functor), getting `.$.field.length`?  I've failed so far to recreate a test for it.

Reply:
Yep, good catch, I'll rewrite it. I think that smth like

```
(ap scalar.$.field field)
(ap field.length length)
```

should trigger it.

Reply:
Yep, I've managed to get `$.field.length` in the debug output both with a field lambda and field lambda + functor, but I was trying to write a full test initially.